daemon/algod/api/server/v2/handlers.go

```diff
@@ -0,0 +1,447 @@
+	}
+
+	err := v2.Node.BroadcastSignedTxGroup(txgroup)
+	if err != nil {
+		return badRequest(ctx, err, err.Error(), v2.Log)
+	}
+
+	// For backwards compatibility, return txid of first tx in group
+	txid := txgroup[0].ID()
+	return ctx.JSON(http.StatusOK, generated.PostTransactionsResponse{TxId: txid.String()})
+}
+
+// TransactionParams returns the suggested parameters for constructing a new transaction.
+// (GET /v2/transactions/params)
+func (v2 *Handlers) TransactionParams(ctx echo.Context) error {
+	stat, err := v2.Node.Status()
+	if err != nil {
+		return internalError(ctx, err, errFailedRetrievingNodeStatus, v2.Log)
+	}
+
+	gh := v2.Node.GenesisHash()
+
+	var params generated.TransactionParametersResponse
+	params.Fee = v2.Node.SuggestedFee().Raw
+	params.GenesisId = v2.Node.GenesisID()
+	params.GenesisHash = gh[:]
+	params.LastRound = uint64(stat.LastRound)
+	params.ConsensusVersion = string(stat.LastVersion)
+
+	proto := config.Consensus[stat.LastVersion]
+	params.MinFee = proto.MinTxnFee
+
+	return ctx.JSON(http.StatusOK, params)
+}
+
+// PendingTransactionInformation gets a specific pending transaction, or looks it up in the ledger if the recently
+// confirmed block is still in the ledger. We keep the last 1000 rounds available in the ledger.
+// (GET /v2/transactions/pending/{txid})
+func (v2 *Handlers) PendingTransactionInformation(ctx echo.Context, txid string, params generated.PendingTransactionInformationParams) error {
+	txID := transactions.Txid{}
+	if err := txID.UnmarshalText([]byte(txid)); err != nil {
+		return badRequest(ctx, err, errNoTxnSpecified, v2.Log)
+	}
+
+	if txn, ok := v2.Node.GetPendingTransaction(txID); ok {
+		response := generated.PendingTransactionResponse{
+			Txn:             "",
+			PoolError:       "",
+			ClosingAmount:   nil,
+			ConfirmedRound:  nil,
+			SenderRewards:   nil,
+			ReceiverRewards: nil,
+			CloseRewards:    nil,
+		}
+
+		handle, err := getCodecHandle(params.Format)
+		if err != nil {
+			return badRequest(ctx, err, errFailedParsingFormatOption, v2.Log)
+		}
+
+		encoded, err := encode(handle, txn.Txn)
+		if err != nil {
+			return internalError(ctx, err, errFailedToParseTransaction, v2.Log)
+		}
+
+		response.Txn = encoded
+
+		if txn.ConfirmedRound != 0 {
+			r := uint64(txn.ConfirmedRound)
+			response.ConfirmedRound = &r
+
+			response.ClosingAmount = &txn.ApplyData.ClosingAmount.Raw
+			response.SenderRewards = &txn.ApplyData.SenderRewards.Raw
+			response.ReceiverRewards = &txn.ApplyData.ReceiverRewards.Raw
+			response.CloseRewards = &txn.ApplyData.CloseRewards.Raw
+
+			response.AssetIndex = computeAssetIndexFromTxn(txn, v2.Node.Ledger())
+		}
+
+		return ctx.JSON(http.StatusOK, response)
+	}
+
+	// We didn't find it, return a failure
+	err := errors.New(errTransactionNotFound)
+	return notFound(ctx, err, err.Error(), v2.Log)
+}
+
+// getPendingTransactions is a generalized version of the get pending transactions endpoints for code reuse.
```

Comment:
consider changing the comment to describe the function by what it does and not by other functions.